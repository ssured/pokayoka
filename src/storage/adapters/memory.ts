import levelup from 'levelup';
import memdown from 'memdown';
import encode from 'encoding-down';
import charwise from 'charwise';
import { SharedAdapter, StorageAdapter, KeyType, ValueType } from './shared';

export class MemoryAdapter extends SharedAdapter implements StorageAdapter {
  constructor() {
    super();
    this.level = levelup(
      encode<KeyType, ValueType>(memdown(), {
        keyEncoding: charwise,
        valueEncoding: 'json',
      })
    );
  }
}
