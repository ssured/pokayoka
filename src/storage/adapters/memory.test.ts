import { MemoryAdapter } from './memory';

describe('MemoryAdapter', () => {
  test('it loads', () => {
    expect(MemoryAdapter).toBeDefined();
  });

  test('it stores and retrieves data', async () => {
    const adp = new MemoryAdapter();

    await adp.batch([
      { type: 'put', key: ['a'], value: 'testa' },
      { type: 'put', key: ['b'], value: 'testb' },
    ]);

    const all = await adp.queryList({});

    expect(all).toMatchSnapshot();
    expect(all.length).toBe(2);

    const data = [];
    for await (const { key, value } of adp.query({})) {
      data.push({ key, value });
    }

    expect(data).toMatchSnapshot();
    expect(data.length).toBe(2);
  });
});
