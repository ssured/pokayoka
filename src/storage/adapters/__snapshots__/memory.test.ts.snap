// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`M it stores and retrieves data 1`] = `
Array [
  Object {
    "key": Array [
      "a",
    ],
    "value": "testa",
  },
  Object {
    "key": Array [
      "b",
    ],
    "value": "testb",
  },
]
`;

exports[`M it stores and retrieves data 2`] = `
Array [
  Object {
    "key": Array [
      "a",
    ],
    "value": "testa",
  },
  Object {
    "key": Array [
      "b",
    ],
    "value": "testb",
  },
]
`;

exports[`Me it stores and retrieves data 1`] = `
Array [
  Object {
    "key": Array [
      "a",
    ],
    "value": "testa",
  },
  Object {
    "key": Array [
      "b",
    ],
    "value": "testb",
  },
]
`;

exports[`Me it stores and retrieves data 2`] = `
Array [
  Object {
    "key": Array [
      "a",
    ],
    "value": "testa",
  },
  Object {
    "key": Array [
      "b",
    ],
    "value": "testb",
  },
]
`;

exports[`Memo it stores and retrieves data 1`] = `
Array [
  Object {
    "key": Array [
      "a",
    ],
    "value": "testa",
  },
  Object {
    "key": Array [
      "b",
    ],
    "value": "testb",
  },
]
`;

exports[`Memo it stores and retrieves data 2`] = `
Array [
  Object {
    "key": Array [
      "a",
    ],
    "value": "testa",
  },
  Object {
    "key": Array [
      "b",
    ],
    "value": "testb",
  },
]
`;

exports[`Memory it stores and retrieves data 1`] = `
Array [
  Object {
    "key": Array [
      "a",
    ],
    "value": "testa",
  },
  Object {
    "key": Array [
      "b",
    ],
    "value": "testb",
  },
]
`;

exports[`Memory it stores and retrieves data 2`] = `
Array [
  Object {
    "key": Array [
      "a",
    ],
    "value": "testa",
  },
  Object {
    "key": Array [
      "b",
    ],
    "value": "testb",
  },
]
`;

exports[`MemoryA it stores and retrieves data 1`] = `
Array [
  Object {
    "key": Array [
      "a",
    ],
    "value": "testa",
  },
  Object {
    "key": Array [
      "b",
    ],
    "value": "testb",
  },
]
`;

exports[`MemoryA it stores and retrieves data 2`] = `
Array [
  Object {
    "key": Array [
      "a",
    ],
    "value": "testa",
  },
  Object {
    "key": Array [
      "b",
    ],
    "value": "testb",
  },
]
`;

exports[`MemoryAda it stores and retrieves data 1`] = `
Array [
  Object {
    "key": Array [
      "a",
    ],
    "value": "testa",
  },
  Object {
    "key": Array [
      "b",
    ],
    "value": "testb",
  },
]
`;

exports[`MemoryAda it stores and retrieves data 2`] = `
Array [
  Object {
    "key": Array [
      "a",
    ],
    "value": "testa",
  },
  Object {
    "key": Array [
      "b",
    ],
    "value": "testb",
  },
]
`;

exports[`MemoryAdap it stores and retrieves data 1`] = `
Array [
  Object {
    "key": Array [
      "a",
    ],
    "value": "testa",
  },
  Object {
    "key": Array [
      "b",
    ],
    "value": "testb",
  },
]
`;

exports[`MemoryAdap it stores and retrieves data 2`] = `
Array [
  Object {
    "key": Array [
      "a",
    ],
    "value": "testa",
  },
  Object {
    "key": Array [
      "b",
    ],
    "value": "testb",
  },
]
`;

exports[`MemoryAdapte it stores and retrieves data 1`] = `
Array [
  Object {
    "key": Array [
      "a",
    ],
    "value": "testa",
  },
  Object {
    "key": Array [
      "b",
    ],
    "value": "testb",
  },
]
`;

exports[`MemoryAdapte it stores and retrieves data 2`] = `
Array [
  Object {
    "key": Array [
      "a",
    ],
    "value": "testa",
  },
  Object {
    "key": Array [
      "b",
    ],
    "value": "testb",
  },
]
`;

exports[`MemoryAdapter it stores and retrieves data 1`] = `
Array [
  Object {
    "key": Array [
      "a",
    ],
    "value": "testa",
  },
  Object {
    "key": Array [
      "b",
    ],
    "value": "testb",
  },
]
`;

exports[`MemoryAdapter it stores and retrieves data 2`] = `
Array [
  Object {
    "key": Array [
      "a",
    ],
    "value": "testa",
  },
  Object {
    "key": Array [
      "b",
    ],
    "value": "testb",
  },
]
`;

exports[`it should work it stores adata 1`] = `
Array [
  Object {
    "key": Array [
      "a",
    ],
    "value": "testa",
  },
  Object {
    "key": Array [
      "b",
    ],
    "value": "testb",
  },
]
`;

exports[`it should work it stores adata 2`] = `
Array [
  Object {
    "key": Array [
      "a",
    ],
    "value": "testa",
  },
  Object {
    "key": Array [
      "b",
    ],
    "value": "testb",
  },
]
`;

exports[`it should work it stores and data 1`] = `
Array [
  Object {
    "key": Array [
      "a",
    ],
    "value": "testa",
  },
  Object {
    "key": Array [
      "b",
    ],
    "value": "testb",
  },
]
`;

exports[`it should work it stores and data 2`] = `
Array [
  Object {
    "key": Array [
      "a",
    ],
    "value": "testa",
  },
  Object {
    "key": Array [
      "b",
    ],
    "value": "testb",
  },
]
`;

exports[`it should work it stores and redata 1`] = `
Array [
  Object {
    "key": Array [
      "a",
    ],
    "value": "testa",
  },
  Object {
    "key": Array [
      "b",
    ],
    "value": "testb",
  },
]
`;

exports[`it should work it stores and redata 2`] = `
Array [
  Object {
    "key": Array [
      "a",
    ],
    "value": "testa",
  },
  Object {
    "key": Array [
      "b",
    ],
    "value": "testb",
  },
]
`;

exports[`it should work it stores and retdata 1`] = `
Array [
  Object {
    "key": Array [
      "a",
    ],
    "value": "testa",
  },
  Object {
    "key": Array [
      "b",
    ],
    "value": "testb",
  },
]
`;

exports[`it should work it stores and retdata 2`] = `
Array [
  Object {
    "key": Array [
      "a",
    ],
    "value": "testa",
  },
  Object {
    "key": Array [
      "b",
    ],
    "value": "testb",
  },
]
`;

exports[`it should work it stores and retrdata 1`] = `
Array [
  Object {
    "key": Array [
      "a",
    ],
    "value": "testa",
  },
  Object {
    "key": Array [
      "b",
    ],
    "value": "testb",
  },
]
`;

exports[`it should work it stores and retrdata 2`] = `
Array [
  Object {
    "key": Array [
      "a",
    ],
    "value": "testa",
  },
  Object {
    "key": Array [
      "b",
    ],
    "value": "testb",
  },
]
`;

exports[`it should work it stores and retriedata 1`] = `
Array [
  Object {
    "key": Array [
      "a",
    ],
    "value": "testa",
  },
  Object {
    "key": Array [
      "b",
    ],
    "value": "testb",
  },
]
`;

exports[`it should work it stores and retriedata 2`] = `
Array [
  Object {
    "key": Array [
      "a",
    ],
    "value": "testa",
  },
  Object {
    "key": Array [
      "b",
    ],
    "value": "testb",
  },
]
`;

exports[`it should work it stores and retrievedata 1`] = `
Array [
  Object {
    "key": Array [
      "a",
    ],
    "value": "testa",
  },
  Object {
    "key": Array [
      "b",
    ],
    "value": "testb",
  },
]
`;

exports[`it should work it stores and retrievedata 2`] = `
Array [
  Object {
    "key": Array [
      "a",
    ],
    "value": "testa",
  },
  Object {
    "key": Array [
      "b",
    ],
    "value": "testb",
  },
]
`;

exports[`it should work it stores and retrieves data 1`] = `
Array [
  Object {
    "key": Array [
      "a",
    ],
    "value": "testa",
  },
  Object {
    "key": Array [
      "b",
    ],
    "value": "testb",
  },
]
`;

exports[`it should work it stores and retrieves data 2`] = `
Array [
  Object {
    "key": Array [
      "a",
    ],
    "value": "testa",
  },
  Object {
    "key": Array [
      "b",
    ],
    "value": "testb",
  },
]
`;

exports[`it should work it stores andata 1`] = `
Array [
  Object {
    "key": Array [
      "a",
    ],
    "value": "testa",
  },
  Object {
    "key": Array [
      "b",
    ],
    "value": "testb",
  },
]
`;

exports[`it should work it stores andata 2`] = `
Array [
  Object {
    "key": Array [
      "a",
    ],
    "value": "testa",
  },
  Object {
    "key": Array [
      "b",
    ],
    "value": "testb",
  },
]
`;

exports[`it should work it stores data 1`] = `
Array [
  Object {
    "key": Array [
      "a",
    ],
    "value": "testa",
  },
  Object {
    "key": Array [
      "b",
    ],
    "value": "testb",
  },
]
`;

exports[`it should work it stores data 2`] = `
Array [
  Object {
    "key": Array [
      "a",
    ],
    "value": "testa",
  },
  Object {
    "key": Array [
      "b",
    ],
    "value": "testb",
  },
]
`;
